### PR TITLE
Add Maker badge board

### PR DESCRIPTION
**Pull request by dronecz**
Thanks🙂

### CODE REVIEW

**Comment by dronecz** [comment]
Failed just at my board, which is expected as my PR is not merged. Am I right or should I look in to it again?

**Comment by dhalbert** [comment]
> Failed just at my board, which is expected as my PR is not merged. Am I right or should I look in to it again?

No, your board should not fail. It looks like the `sdkconfig` file is missing: see https://github.com/adafruit/circuitpython/actions/runs/3357317522/jobs/5563100639.

But there are far too many changed files here. Don't add the frozen libraries directly by copying. Add them as git submodules.

**Comment by dronecz** [comment]
Still unhappy about something 😕

```
Note: Not building languages {'hi', 'ko', 'cs', 'el'}
building boards with parallelism 2
Error: Build maker_badge for de_DE took 0.66s and failed
make: Entering directory '/home/runner/work/circuitpython/circuitpython/ports/espressif'
Use make V=1, make V=2 or set BUILD_VERBOSE similarly in your environment to increase build verbosity.
make: *** No rule to make target 'boards/maker_badge/sdkconfig', needed by 'build-maker_badge/esp-idf/config/sdkconfig.h'.  Stop.
make: *** Waiting for unfinished jobs....
mkdir -p build-maker_badge/genhdr
make: Leaving directory '/home/runner/work/circuitpython/circuitpython/ports/espressif'

Cannot find file ../ports/espressif/build-maker_badge/firmware.uf2
```

Should I add localy build `firmware.uf2` file?

File: ports/espressif/boards/maker_badge/mpconfigboard.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+#define MICROPY_HW_BOARD_NAME       "Maker badge by Czech maker"
+#define MICROPY_HW_MCU_NAME         "ESP32S2"
+
+#define MICROPY_HW_NEOPIXEL (&pin_GPIO18)
+#define MICROPY_HW_NEOPIXEL_COUNT (4)
+
+#define CIRCUITPY_BOOT_BUTTON (&pin_GPIO0)
+
+#define BOARD_USER_SAFE_MODE_ACTION translate("pressing boot button at start up.\n")
```
Comment (jepler) [inline ports/espressif/boards/maker_badge/mpconfigboard.h:37]: The safe mode messages were recently re-worked. As a result, this line is not needed. Please remove this line. When CIRCUITPY_BOOT_BUTTON is defined, a predefined message is used:
```c
            message = translate("The BOOT button was pressed at start up.\n");
```

File: ports/espressif/boards/maker_badge/mpconfigboard.mk [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# so increase it to 32.
+CFLAGS += -DCFG_TUD_TASK_QUEUE_SZ=32
+
+CIRCUITPY_ESP32_CAMERA = 0
+
+CIRCUITPY_ESP_FLASH_MODE=dio
+CIRCUITPY_ESP_FLASH_FREQ=40m
+CIRCUITPY_ESP_FLASH_SIZE=4MB
+
+CIRCUITPY_MODULE=wroom
```
Comment (jepler) [inline ports/espressif/boards/maker_badge/mpconfigboard.mk:21]: CIRCUITPY_MODULE is no longer used. Remove this line.

File: ports/espressif/boards/maker_badge/mpconfigboard.mk [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+USB_MANUFACTURER = "Czech maker"
+
+IDF_TARGET = esp32s2
+
+INTERNAL_FLASH_FILESYSTEM = 1
+LONGINT_IMPL = MPZ
+
+# The default queue depth of 16 overflows on release builds,
+# so increase it to 32.
+CFLAGS += -DCFG_TUD_TASK_QUEUE_SZ=32
```
Comment (jepler) [inline ports/espressif/boards/maker_badge/mpconfigboard.mk:13]: These lines are not needed, because these are defaults on espressif.  Remove them please.

**Review by jepler — COMMENTED** [review]
This pull request isn't building because espressif boards each need an "sdkconfig" file. The specific contents of the sdkconfig file depend on the espressif chip / module used.

Not all sdkconfig settings have to be in this file, but these do:
 * the size and speed of spiram
 * psram pinout information, in some cases

For example, this is the sdkconfig file for the kaluga devkit: https://github.com/adafruit/circuitpython/blob/main/ports/espressif/boards/espressif_kaluga_1.3/sdkconfig

I'm not 100% sure of the details of this file.

**Comment by jepler** [comment]
Hi and thanks for the PR! I have some comments / requested changes. I'm sorry that I don't know enough about the sdkconfig file to guide you through getting the right content into it.

File: ports/espressif/boards/maker_badge/mpconfigboard.mk (hunk 3 above)
Comment (dronecz) [reply ports/espressif/boards/maker_badge/mpconfigboard.mk]: Thanks, I removed those lines 🙂

File: ports/espressif/boards/maker_badge/mpconfigboard.mk (hunk 2 above)
Comment (dronecz) [reply ports/espressif/boards/maker_badge/mpconfigboard.mk:12]: Thanks, I removed that line 🙂

File: ports/espressif/boards/maker_badge/mpconfigboard.h (hunk 1 above)
Comment (dronecz) [reply ports/espressif/boards/maker_badge/mpconfigboard.h]: Thanks, I removed that line 🙂

**Comment by dronecz** [comment]
Thank you @jepler I could build localy so I was confused, why this build failed. I had `sdkconfig` file localy, but I do not know why, but I did not copy it to PR 🤦‍♂️. I hope this will build.

**Comment by jepler** [comment]
The failures you're seeing now are due to an incompatibility of esp-idf and python 3.11, which has started rolling out on github actions. I filed a PR to pin our python back at 3.10 during the build: https://github.com/adafruit/circuitpython/pull/7164

**Comment by dronecz** [comment]
> The failures you're seeing now are due to an incompatibility of esp-idf and python 3.11, which has started rolling out on github actions. I filed a PR to pin our python back at 3.10 during the build: #7164

Thanks 🙂 Is there way, how can I restart build without making changes to my code?

**Comment by jepler** [comment]
I updated the PR to remove the change to circuitpython.pot, which is unneeded now anyway. This should cause a fresh re-builld with the fix to the python 3.11 problem too.

File: ports/espressif/boards/maker_badge/sdkonfig [hunk 4]
```diff
@@ -2,5 +2,5 @@ CONFIG_ESP32S2_SPIRAM_SUPPORT=n
 
 # LWIP
 #
-CONFIG_LWIP_LOCAL_HOSTNAME="espressif"
+CONFIG_LWIP_LOCAL_HOSTNAME="Maker_badge"
```
Comment (jepler) [inline ports/espressif/boards/maker_badge/sdkonfig:5]: The filename is misspelled

Comment (dronecz) [reply ports/espressif/boards/maker_badge/sdkonfig]: Thank you! I missed that. 😩

**Comment by dronecz** [comment]
@jepler It is again failing at language files. I really do not need `circuitpython.pot` file? Thanks

**Comment by jepler** [comment]
The build is failing for the following reason:
```
make: *** No rule to make target '../../frozen/Adafruit_CircuitPython_UC8151D', needed by 'build-maker_badge/frozen_mpy'.  Stop.
```

Additional steps are needed to properly add a new frozen module that has never been included before in any board. We have a guide on doing this via git on your local computer: https://learn.adafruit.com/building-circuitpython/adding-frozen-modules -- as far as I know you can't add a new submodule merely by the github website interface.

**Comment by jepler** [comment]
The build failed for the following reason:
```
subprocess.CalledProcessError: Command 'git describe --tags --exact-match' returned non-zero exit status 128.
```
as a policy, each frozen submodule must point exactly at a tag (release), not at another ref. There's a script for updating all libraries. I ran it and updated your branch, so maybe this time it will build successfully.

**Comment by dronecz** [comment]
@jepler Many thanks for your help! I tried to fix it yesterday at my local build, but without success. Do you have more info about that script to fix this issue? I saw this information at page, which you linked earlier, but I did not figure out how to make it work.
 
image

It would be good to mention that script in the learn guide, so peple will not strugle to fix same issue.

**Comment by jepler** [comment]
The command is "make update-frozen-libraries" in the top level directory. I left feedback on the guide using the " Feedback? Corrections?" link so hopefully the maintainer will incorporate the suggestion soon.

**Review by jepler — APPROVED** [review]
Thanks!  It's taken a bit of time and I appreciate your responsiveness to my questions and suggestions. I think it's ready now.

**Comment by dhalbert** [comment]
> I left feedback on the guide using the " Feedback? Corrections?" link so hopefully the maintainer will incorporate the suggestion soon.

Done.

**Pull request merged**